packages/hash/frontend/src/components/RemoteBlock/blockRenderer/customElement.tsx

```diff
@@ -0,0 +1,56 @@
+import { createComponent } from "@lit-labs/react";
+import React, { useMemo, VFC } from "react";
+
+type CustomElementLoaderProps = {
+  elementClass: typeof HTMLElement;
+  properties: Record<string, unknown>;
+  tagName: string;
+};
+
+/**
+ * Registers (if not already registered) and loads a custom element.
+ */
+export const CustomElementLoader: VFC<CustomElementLoaderProps> = ({
+  elementClass,
+  properties,
+  tagName,
+}) => {
+  /**
+   * Register the element with the CustomElementsRegistry, if not already registered.
+   */
+  let existingCustomElement = customElements.get(tagName);
+  if (!existingCustomElement) {
+    try {
+      customElements.define(tagName, elementClass);
```

Comment:
Thanks for the flag and example - had a go at this in https://github.com/hashintel/hash/pull/752/commits/d22af8b1c1f4e692cc9434e215d43221ce1aaaa1

Only issue is that I had to cast the return of `createComponent` to `any` at the bottom because I was getting "Expression produces a union type that is too complex to represent". It was previously implicitly `any`. I think this is fine for now unless you have any advice on what to do about it.